src/main/java/org/elasticsearch/index/mapper/internal/IdFieldMapper.java

```diff
@@ -144,6 +146,8 @@ protected IdFieldMapper(String name, String indexName, float boost, FieldType fi
         super(new Names(name, indexName, indexName, name), boost, fieldType, docValues, Lucene.KEYWORD_ANALYZER,
                 Lucene.KEYWORD_ANALYZER, postingsProvider, docValuesProvider, null, null, fieldDataSettings, indexSettings);
         this.path = path;
+        this.writePre20Settings = indexSettings.equals(ImmutableSettings.EMPTY) == false &&
```

Comment:
Since all 1.x versions will have the created version in the index settings, the empty settings check can be removed.